src/librustdoc/html/render/print_item.rs

```diff
@@ -1618,6 +1619,9 @@ fn item_variants(
         if let clean::VariantKind::Tuple(ref s) = variant_data.kind {
             write!(w, "({})", print_tuple_struct_fields(cx, s));
         }
+        if variant.attrs.other_attrs.iter().any(|a| a.has_name(kw::Default)) {
+            w.write_str(&chip("Default", "default"));
```

Comment:
Not sure if it's worth it but you could update `chip` to return `fmt::Arguments<'_>` to avoid a `String` allocation similar to #117007. Not sure how hot this code path is though :shrug:

```suggestion
            w.write_fmt(chip("Default", "default"));
```

Reply:
I tried with `fmt::Arguments<'a>` and `display_fn`, couldn't make it work.

I'm not sure if it is worth it or not, too. The string allocation is not big in this case and also adding that much complexity there may not worth it.